klippa/src/post.rs

```diff
@@ -0,0 +1,128 @@
+//! impl subset() for post
+
+use std::collections::HashMap;
+
+use crate::{Plan, SubsetError, SubsetError::SubsetTableError, SubsetFlags};
+use write_fonts::{
+    read::{
+        tables::post::{Post, DEFAULT_GLYPH_NAMES},
+        FontRef, TableProvider, TopLevelTable,
+    },
+    types::GlyphId,
+    FontBuilder,
+};
+
+// reference: subset() for post in harfbuzz
+// https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-ot-post-table.hh#L96
+pub(crate) fn subset_post(
+    font: &FontRef,
+    plan: &Plan,
+    builder: &mut FontBuilder,
+) -> Result<(), SubsetError> {
+    let post = font.post().or(Err(SubsetTableError(Post::TAG)))?;
+    let mut out = Vec::with_capacity(post.offset_data().len());
+    // copy header
+    out.extend_from_slice(post.offset_data().as_bytes().get(0..32).unwrap());
```

Comment:
looking at this pattern happening everywhere, I wanted to point you to the `BeBuffer` type in read-fonts; it's intended for generating test cases, but perhaps it could be a useful demonstration of a more ergonomic way of expressing this pattern? https://github.com/googlefonts/fontations/blob/main/read-fonts/src/tests/test_helpers.rs

Reply:
Thanks!